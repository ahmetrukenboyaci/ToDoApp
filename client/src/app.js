/** Dependencies **/
import React, {useCallback, useEffect, useState} from 'react';

/** Components **/
import ToDoInput from './components/ToDoInput/ToDoInput';
import ToDoList from './components/ToDoList/ToDoList';
import Button from './components/Button/Button';

/** Utils **/
import * as types from './utils/methodTypes';
import {request} from './utils/request';

/** Styles **/
import styles from './app.module.scss';

/** Images **/
import AddIcon from './assets/images/AddIcon';

const App = () => {
	const [toDos, setToDos] = useState([]);
	const [newTodoValue, setNewTodoValue] = useState('');

	const onTodoValueChange = useCallback((value) => {
		setNewTodoValue(value);
	}, []);

	const getToDos = useCallback((isPostOrPut=false) => {
		request('/todos', types.GET).then((r) => {
			setToDos(r);
			if (isPostOrPut)
				setNewTodoValue('');
		});
	},[]);

	const addNewTodo = useCallback(() => {
		if (newTodoValue.length) {
			const data = {
				description: newTodoValue
			};

			request('/todos', types.POST, data).then(() => getToDos(true));
		}
	}, [getToDos, newTodoValue]);

	const onDelete = useCallback((id) => {
		request(`/todos/${id}`, types.DELETE).then(() => getToDos());
	},[getToDos]);

	const onComplete = useCallback((id, done) => {
		const data = {
			done: !done
		};

		request(`/todos/${id}`, types.PUT, data).then(() => getToDos(true));
	},[getToDos]);

	const onClearAll = useCallback(() => {
		request('/todos', types.DELETE).then(() => getToDos());
	},[getToDos]);

	useEffect(() => {
		getToDos();
	}, [getToDos]);

	return (
		<div className={styles.app}>
			<div className={styles.header}>To Dos</div>
			<div className={styles.addNewToDo}>
				<ToDoInput onChange={onTodoValueChange} onKeyDown={addNewTodo} newValue={newTodoValue} />
				<div id={'add-icon'} className={styles.addIcon} onClick={addNewTodo}>
					<AddIcon height={50}/>
				</div>
			</div>
			<ToDoList data={toDos} onDelete={onDelete} onComplete={onComplete} />
			<div className={styles.footer}>
				<Button text={'Clear All'} onClick={onClearAll}/>
			</div>
		</div>
	);
};

export default App;