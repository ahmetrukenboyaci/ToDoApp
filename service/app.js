const express = require("express");
const cors = require("cors");

const todos = require("./routes/todos");

const connection = require("./db");

const app = express();
const port = process.env.PORT || 5000;

connection.dbconnect();

app.use(express.json());
app.use(cors());

app.use("/api/todos", todos);

app.listen(port, () => console.log(`Listening on port ${port}...`))

module.exports = app;