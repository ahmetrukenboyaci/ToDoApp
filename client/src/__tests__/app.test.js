import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import {configure, shallow} from 'enzyme';
import app from '../app';

configure({adapter: new Adapter()});

describe('app', () => {
	it('should render app component', () => {
		// eslint-disable-next-line no-unused-vars
		const wrapper = shallow(<app/>);
	});

	it('should render initial layout', () => {

		const component = shallow(<app/>);

		expect(component.getElements()).toMatchSnapshot();
	});
});