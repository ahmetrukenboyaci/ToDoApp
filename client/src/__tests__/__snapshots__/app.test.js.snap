// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app should render initial layout 1`] = `
Array [
  <app />,
]
`;
